include/vcpu/arch/intel_x64/vcpu.h

```diff
@@ -36,14 +39,32 @@ class vcpu : public bfvmm::intel_x64::vcpu
 {
 public:
 
+    using handler_list_t = std::unordered_map<vmcs_n::value_type, base *>;
+    const static std::list<vmcs_n::value_type> s_available_handlers;
+
     /// Default Constructor
     ///
     /// @expects
     /// @ensures
     ///
     vcpu(vcpuid::type id) :
-        bfvmm::intel_x64::vcpu{id}
-    { }
+        bfvmm::intel_x64::vcpu{id},
+        m_used_handlers{&s_available_handlers}
+    {
+        init_handlers();
+    }
+
+    /// Constructor with specified handlers
+    ///
+    /// @expects
+    /// @ensures
+    ///
+    vcpu(vcpuid::type id, std::list<vmcs_n::value_type> *needed_handlers) :
```

Comment:
Not a big fan of this. If you want a handler, you need to enable it, and then the enable function creates the handler.